src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -720,37 +720,30 @@ static int CountDigits(int value)
         // extended attributes. They get collected and saved in that dictionary, with no restrictions.
         private void CollectExtendedAttributesFromStandardFieldsIfNeeded()
         {
-            ExtendedAttributes.Add(PaxEaName, _name);
+            ExtendedAttributes[PaxEaName] = _name;
 
             if (!ExtendedAttributes.ContainsKey(PaxEaMTime))
             {
                 ExtendedAttributes.Add(PaxEaMTime, TarHelpers.GetTimestampStringFromDateTimeOffset(_mTime));
             }
```

Comment:
Just pointing out some other issues in this code:

1. someone uses the copy ctor. passing the extended attributes from the other entry.
2. on the new entry, you set ModificationTime.
3. pass the new entry to TarWriter.WriteEntry.

The modification time will be neglected due to this check. @carlossanlop please advice if this is something we should address in this PR.

Reply:
> please advice if this is something we should address in this PR.

Yes. We must always update the dictionary value for mtime using the value from the ModificationTime property. So the if condition should be removed.

Reply:
Do you recall why was added in the first place?